### PR TITLE
Allows the Mime to communicate via Ventriloquist Dummy

### PR DESCRIPTION
**Pull request by Tranquill6**
[Guidelines]: # Good enough.

[Changelogs]: # Adds a spell that Mimes start with that enable it to communicate while holding the Ventriloquist Dummy in their hand. Relates to issue #9647 

[why]: This allows the Mime, who wishes not to break the vow of silence, a method of communication to our lovely, not special, crew members.

### CODE REVIEW

**Review by yoyobatty — DISMISSED** [review]
Code seems good to me

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -35,6 +35,9 @@
 /*
  * Balloons
  */
+
+ GLOBAL_VAR_INIT(doll_name, "Dummy")
```
Comment (k-k-karma) [inline code/game/objects/items/toys.dm]: wrong section and why a global var

Comment (Tranquill6) [reply code/game/objects/items/toys.dm]: It's fine, the global var is so I can pull up the name of the doll in the mime's spell dm.

Comment (k-k-karma) [reply code/game/objects/items/toys.dm]: or just use .name?
global vars are overused

File: code/modules/spells/spell_types/mime.dm [hunk 2]
```diff
@@ -128,6 +128,36 @@
 		invocation_type ="none"
 	..()
 
+// This spell enables muted Mimes to use the Ventriloquist Dummy to communicate
```
Comment (k-k-karma) [inline code/modules/spells/spell_types/mime.dm]: modularize please

File: code/modules/spells/spell_types/mime.dm [hunk 3]
```diff
@@ -128,6 +128,36 @@
...
+	action_background_icon_state = "bg_mime"
+
+/obj/effect/proc_holder/spell/targeted/mime/dummy_talk_into/Click(atom/movable/A)
+	var/obj/item/I = usr.is_holding_item_of_type(/obj/item/toy/dummy)
+	if(I)
+		var/input = stripped_input(usr, "Please enter a message to say through the dummy.", "Ventriloquist Dummy", "")
+		var/mob/M = A
+		if (istype(M))
+			M.log_talk(message, LOG_SAY, tag="dummy toy")
+		I.say(input)
```
Comment (k-k-karma) [inline code/modules/spells/spell_types/mime.dm]: I.talk_into(input)

File: code/modules/spells/spell_types/mime.dm (hunk 2 above)
Comment (yoyobatty) [reply code/modules/spells/spell_types/mime.dm]: you mean hippie comment, didn't see this oops

File: code/modules/spells/spell_types/mime.dm [hunk 4]
```diff
@@ -128,6 +128,36 @@
...
+	sound = null
+	human_req = TRUE
+	charge_max = 3000
+	range = -1
+	include_user = TRUE
+
+	action_icon_state = "mime"
+	action_background_icon_state = "bg_mime"
+
+/obj/effect/proc_holder/spell/targeted/mime/dummy_talk_into/Click(atom/movable/A)
```
Comment (yoyobatty) [inline code/modules/spells/spell_types/mime.dm]: Remove the argument `atom/movable/A`

Comment (yoyobatty) [reply code/modules/spells/spell_types/mime.dm]: Insert `var/mob/living/carbon/human/owner = usr` line below

File: code/modules/spells/spell_types/mime.dm [hunk 5]
```diff
@@ -128,6 +128,36 @@
...
+	var/obj/item/I = usr.is_holding_item_of_type(/obj/item/toy/dummy)
```
Comment (yoyobatty) [inline code/modules/spells/spell_types/mime.dm]: replace usr with owner

File: code/modules/spells/spell_types/mime.dm [hunk 6]
```diff
@@ -128,6 +128,36 @@
...
+	if(I)
+		var/input = stripped_input(usr, "Please enter a message to say through the dummy.", "Ventriloquist Dummy", "")
+		var/mob/M = A
```
Comment (yoyobatty) [inline code/modules/spells/spell_types/mime.dm]: remove this line

File: code/modules/spells/spell_types/mime.dm [hunk 7]
```diff
@@ -128,6 +128,36 @@
...
+		if (istype(M))
```
Comment (yoyobatty) [inline code/modules/spells/spell_types/mime.dm]: remove this line

File: code/modules/spells/spell_types/mime.dm [hunk 8]
```diff
@@ -128,6 +128,36 @@
...
+			M.log_talk(message, LOG_SAY, tag="dummy toy")
```
Comment (yoyobatty) [inline code/modules/spells/spell_types/mime.dm]: replace M with `owner`

**Review by yoyobatty — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/mime_dummy.dm [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+	action_icon_state = "mime"
+	action_background_icon_state = "bg_mime"
+
+/obj/effect/proc_holder/spell/targeted/mime/dummy_talk_into/Click()
+	var/mob/living/carbon/human/owner = usr
+	var/obj/item/I = owner.is_holding_item_of_type(/obj/item/toy/dummy)
+	if(I)
+		var/input = stripped_input(usr, "Please enter a message to say through the dummy.", "Ventriloquist Dummy", "")
+		owner.log_talk(message, LOG_SAY, tag="dummy toy")
+		I.say(input)
```
Comment (k-k-karma) [inline code/modules/spells/spell_types/mime_dummy.dm]:
```suggestion
		I.say_into(input)
```

File: code/modules/spells/spell_types/mime_dummy.dm [hunk 10]
```diff
@@ -0,0 +1,26 @@
+// This spell enables muted Mimes to use the Ventriloquist Dummy to communicate
```
Comment (k-k-karma) [inline code/modules/spells/spell_types/mime_dummy.dm]: To modularize, you must put your code where your code would originally go on TG, but in the hippiestation folder

Comment (yoyobatty) [reply code/modules/spells/spell_types/mime_dummy.dm]: code/modules/spells/spell_types/mime_dummy.dm = tg
hippiestation/code/modules/spells/spell_types/mime_dummy.dm = hippie modularized

**Comment by bardofrage** [comment]
eh +1

**Comment by DaydreamPuppeh** [comment]
+1

**Comment by yoyobatty** [comment]
wut?

**Comment by DaydreamPuppeh** [comment]
Meant to say. Why was this closed

**Comment by yoyobatty** [comment]
I actually don't know, I thought this got remade?

**Pull request closed without merging**